VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioComponent.cpp

```diff
@@ -790,6 +833,36 @@ void UVitruvioComponent::Initialize()
 	{
 		PropertyChangeDelegate = FCoreUObjectDelegates::OnObjectPropertyChanged.AddUObject(this, &UVitruvioComponent::OnPropertyChanged);
 	}
+	
+	auto ActorMoved = [this](const AActor* Actor)
+	{
+		if (Actor != GetOwner())
+		{
+			return;
+		}
+		
+		if (UVitruvioComponent* VitruvioComponent = Actor->FindComponentByClass<UVitruvioComponent>())
+		{
+			if (!VitruvioComponent->IsBatchGenerated() && VitruvioComponent->bEnableOcclusionQueries && VitruvioComponent->GenerateAutomatically)
+			{
+				VitruvioComponent->Generate();
+			}
+		}
+	};
+	
+	OnActorMoved = GEngine->OnActorMoved().AddLambda([ActorMoved](AActor* Actor)
+	{
+		ActorMoved(Actor);
+	});
+
+	OnActorsMoved = GEngine->OnActorsMoved().AddLambda([ActorMoved](const TArray<AActor*>& Actors)
+	{
+		for (const AActor* Actor : Actors)
+		{
+			ActorMoved(Actor);
+		}
+	});
+
```

Comment:
Maybe if would be better to use the `FTransformUpdated TransformUpdated;` event on the component instead of the global event.